python/ray/tune/result_grid.py

```diff
@@ -0,0 +1,37 @@
+from ray.ml.result import Result
+from ray.tune import ExperimentAnalysis
+from ray.tune.trial import Trial
+from ray.util import PublicAPI
+
+
+# TODO(xwjiang): Change to alpha
+@PublicAPI(stability="beta")
+class ResultGrid:
+    """A summary returned by ``tuner.fit()`` call.
+
+    You can use it to inspect the trials run as well as obtaining the best result.
```

Comment:
Comment on how failed trials are handled? E.g., is there an error result?

Would be great to have a few code examples too.

Reply:
Hmmm, good point. 
Just from the name of it (ResultGrid), I am a little hesitant to pack all the information in it (including exception, failed trials, path to the restore file, etc). I wish it could be just what the name implies.

So here is my proposal.
ResultGrid is just to present results from terminated trials. 
If user wants to inspect errored trials, there is this nicely formatted table like
```
number of errored trials: 1
trial.   |.      fail number     |.           error file
```
In case an exception happens in driver process that the whole thing crashes, then the console will output
```
... stack trace ...
RuntimeError("this is a simulated error!")
Please use tuner = Tuner.restore("/Users/xwjiang/ray_results/tuner_resume") to resume.
```

Ultimately there are 3 places that user can consume information from:
1. the object as defined by our public API
2. console output
3. files

Would like to know how you rank them in terms of usability.

Reply:
I think one key user feedback in the past was they would still like to access their results even if some trials failed. For example, if only 1 trial of 1000 failed.

This is important enough that I think it should be in the v0 of this API. Maybe can be something simple, like an error / exception flag?

Reply:
hmmm, are you proposing 
```
Result {
status:  # status of the corresponding trials: TERMINATED, NON_TERMINATED, ERROR
checkpoint:
metrics:
}
```
For the 1 failed trial, checkpoint and metrics will be the last one as perceived by Tune driver component. Does that sound good?
In this way, we can present trials of all status in result grid.

Reply:
Added both status and error. Updated doc. PTAL.

Reply:
Actually removed status as it's a little redundant. Just error is needed.

Reply:
Will user know the error type?

Reply:
TuneError for now. 
Later we should have more granular TuneError subclasses for different types of errors.